ui/src/app/utils/machines.ts

```diff
@@ -50,6 +57,203 @@ const machineBases: Machine[] = [
   { name: 'n1-highcpu-96', cpu: 96, memory: 86.4, price: 3.402, preemptiblePrice: 0.7200 }
 ];
 
+export interface Gpu {
+  machineName: string;
+  gpuType: gpuTypes;
+  gpuNum: number;
+  gpuPrice: number;
+}
+
+export const gpuBases: Gpu[] = [
```

Comment:
Check how Terra-ui implement this: 
https://github.com/DataBiosphere/terra-ui/blob/dev/src/data/machines.js#L35
1: I think the price might be just sum existing things + https://cloud.google.com/compute/gpus-pricing
(code: https://github.com/DataBiosphere/terra-ui/blob/dev/src/libs/runtime-utils.js#L92-L100)
2: Also, they use `maxNumCpus` and `maxMem` to decide the validate GCP list instead of listing all possible combinations between GPU and CPU.
3: I am not sure the exact detail how they implement this, but I know code is:
https://github.com/DataBiosphere/terra-ui/blob/dev/src/libs/runtime-utils.js#L55
https://github.com/DataBiosphere/terra-ui/blob/dev/src/components/ComputeModal.js#L703
Can we do something similar?

Reply:
Thanks! I think that should be the inner logic that terra does. I didn't notice that. I enumerated them since It's confusing when I first saw the non-linear price table.